relentless/potential/potential.py

```diff
@@ -34,28 +123,91 @@ def evaluate(self, pair):
                 params[p] = v
             elif v is not None:
                 raise TypeError('Parameter type unrecognized')
-
             if v is None:
                 raise ValueError('Parameter {} is not set for ({},{}).'.format(p,pair[0],pair[1]))
 
         return params
 
+    def update(self, **values):
+        """Partially reassigns coefficient values.
+
+        Sets coefficient parameter values. Other parameter values are not reset to the default.
+
+        Parameters
+        ----------
+        values : `dict` or "keyword args"
+            The parameter values to be updated/over-written.
+
+        Raises
+        ------
+        KeyError
+            If parameter to be updated is not in the coefficient matrix.
+
+        """
+        if isinstance(values, dict):
+            for key in values:
+                self[key] = values[key]
+        else:
+            for key,value in values.items():
+                key = str(key)
+                if key not in self.params:
+                    raise KeyError('Only the known parameters can be set in the coefficient matrix.')
+                self[key] = value
+
```

Comment:
Sorry I didn't notice this earlier. This method needs to be defined for the `FixedKeyDict`, not the `CoefficientMatrix` since we want to be able to do
```
m['A','B'].update(coeffs)
```
which is calling the `update` method of `m['A','B']`. Can you move it over to that class?

Also, I'm not sure if this call signature is quite right. I think you need to accept an optional positional argument and optional keyword args, like this:
```py
def update(self, *other, **values):
    if len(other) > 1:
        # raise an error if more than one positional arg is given

    # use other[0] if given, otherwise try to use the optional kwargs
    d = other[0] if len(other) == 1 else values

    # iterate over d and assign values using the second code path
    for key,value in d.items():
        # set stuff as before
```